controllers/components/kserve/kserve_support.go

```diff
@@ -153,7 +154,11 @@ func removeServerlessFeatures(ctx context.Context, cli client.Client, k *compone
 
 func getDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) (string, error) {
 	kserveConfigMap := corev1.ConfigMap{}
-	if err := cli.Get(ctx, client.ObjectKey{Name: kserveConfigMapName, Namespace: dscispec.ApplicationsNamespace}, &kserveConfigMap); err != nil {
+	err := cli.Get(ctx, client.ObjectKey{Name: kserveConfigMapName, Namespace: dscispec.ApplicationsNamespace}, &kserveConfigMap)
+	if errors.IsNotFound(err) {
+		return "", nil
```

Comment:
this will set k.Statu.sDefaultDeploymentMode to "" ?
or should it `return string(Serverless), nil`